### PR TITLE
Change minimum supported node to 12

### PR DESCRIPTION
**Pull request by pastelsky**
**Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
Node versions as low as v10 are supported and babel-loader's source code is transpiled heavily to suit older node versions.

**What is the new behavior?**
Upgrades minimum node version requirement to v12. v12 itself is in maintenance mode and due to be outdated soon — it doesn't make a lot of sense to keep supporting node 10 or below.

Additionally, I've upgraded yarn and regenerated yarn.lock too.

**Does this PR introduce a breaking change?**
- [x] Yes
- [ ] No

If this PR contains a breaking change, please describe the following...

* Impact: Users using node < 12 will no longer be able to run babel-loader
* Migration path for existing applications: Upgrade to a maintained node version — like Node 12 or above

### CODE REVIEW

**Comment by pastelsky** [comment]
This unlocks usage of https://github.com/babel/babel-loader/pull/932

**Comment by hussinhassan80** [comment]
Files changed.

File: package.json [hunk 1]
```diff
@@ -11,8 +11,8 @@
   },
   "dependencies": {
     "find-cache-dir": "^3.3.1",
+    "lmdb": "^2.2.4",
```
Comment (JLHwung) [inline package.json:14]: The `package.json` changes are part of #932, can you revert them and update the lockfile accordingly for this PR?

File: babel.config.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
     ["@babel/preset-env", {
       "loose": true,
       "targets": {
-        "node": "6.9"
+        "node": "12.0.0"
```
Comment (JLHwung) [inline babel.config.json:6]: nit: Let's bump to `12.17.0`, the first node 12 version with ESM unflagged.

**Pull request closed without merging**